sqtom/fitting_1d.py

```diff
@@ -25,23 +25,90 @@
 """
 
 
+import warnings
 import numpy as np
+from scipy.optimize import root_scalar
 from lmfit import Minimizer, Parameters
 from sqtom.forward_solver import degenerate_pmf
 
 
+def two_schmidt_mode_guess(pd_data, sq_label="sq_", noise_fraction=0.001, atol=1e-8):
+    """Given a single mode histogram, this function generates a "physically" motivated guess for the loss, Schmidt
+    occupations and dark counts parameters.
+
+    This model is sensible only if there is a probability for an n == 2 event. If there are only probabilities for
+    n == 0 and n == 1 then the g2 will erroneously be calculated as 0 and as a result the remainder of the estimtes will
+    be nonsense.
+
+    Args:
+        pd_data (array): rectangular array with the probability mass functions of the photon events
+        sq_label (string): label for the squeezing parameters.
+
+    Returns:
+        dict: dictionary containing a set of "reasonable" model parameters. In the case that the pd_data only has
+        entries for n == 0, or n == 0 and n == 1, it returns None along with a warning message.
+    """
+    res = marginal_calcs_1d(pd_data)
+    nmean = res["n"]
+    g2 = res["g2"]
+    P0 = pd_data[0]
+    if np.allclose(nmean, 0, atol=atol):
+        warnings.warn("The average photon number is zero")
+        return None
+    if np.allclose(g2, 0, atol=atol):
+        warnings.warn("Not enough samples for suitable guess")
+        return None
+
+    def findeta(eta, nmean, g2, P0):
```

Comment:
(Minor suggestion): could be worth separating the two words with an underscore here. If agreed, care would be taken for other calls to this function (they'd need to be adjusted too).
```suggestion
    def find_eta(eta, nmean, g2, P0):
```

Reply:
Good idea. Thanks!